packages/venia-concept/src/components/ProductFullDetail/ProductFullDetail.js

```diff
@@ -1,254 +1,241 @@
-                    </Button>
-                </section>
-                <section className={classes.description}>
-                    <h2 className={classes.descriptionTitle}>
-                        <span>Product Description</span>
-                    </h2>
-                    <RichText content={product.description} />
-                </section>
-                <section className={classes.details}>
-                    <h2 className={classes.detailsTitle}>
-                        <span>SKU</span>
-                    </h2>
-                    <strong>{product.sku}</strong>
-                </section>
-            </Form>
-        );
-    }
-}
+    const handleSelectionChange = useCallback(
+        (optionId, selection) => {
+            // We must create a new Map here so that React knows that the value
+            // of optionSelections has changed.
+            const newOptionSelections = new Map([...optionSelections]);
+            newOptionSelections.set(optionId, Array.from(selection).pop());
+            setOptionSelections(newOptionSelections);
+        },
+        [optionSelections]
+    );
+
+    return (
+        <Suspense fallback={loadingIndicator}>
+            <Options
+                options={configurable_options}
+                onSelectionChange={handleSelectionChange}
+            />
+        </Suspense>
+    );
+};
+
+const ProductFullDetail = props => {
+    // Props.
+    const { addToCart, isAddingItem, product } = props;
+
+    // State.
+    const [quantity, setQuantity] = useState(INITIAL_QUANTITY);
+    const [optionSelections, setOptionSelections] = useState(
+        INITIAL_OPTION_SELECTIONS
+    );
+    const derivedOptionCodes = deriveOptionCodesFromProduct(product);
+    const [optionCodes] = useState(derivedOptionCodes);
+
+    // Members.
+    const { amount: productPrice } = product.price.regularPrice;
+    const classes = mergeClasses(defaultClasses, props.classes);
+    const isMissingOptions = getIsMissingOptions(product, optionSelections);
+    const mediaGalleryEntries = getMediaGalleryEntries(
+        product,
+        optionCodes,
+        optionSelections
+    );
+    const productOptions = getProductOptions(
+        product,
+        optionSelections,
+        setOptionSelections
+    );
+
+    // Event handlers.
+    const handleAddToCart = useCallback(() => {
+        const payload = {
+            item: product,
+            productType: product.__typename,
+            quantity
+        };
+
+        if (isProductConfigurable(product)) {
+            appendOptionsToPayload(payload, optionSelections, optionCodes);
+        }
 
-export default classify(defaultClasses)(ProductFullDetail);
+        addToCart(payload);
+    }, [addToCart, optionCodes, optionSelections, product, quantity]);
+
+    return (
+        <Form className={classes.root}>
+            <section className={classes.title}>
+                <h1 className={classes.productName}>{product.name}</h1>
```

Comment:
Pointing out that heading tags no longer nest their contents inside of a `span`.  This used to look like:

```jsx
<h1 className={classes.productName}>
  <span>{product.name}</span>
</h1>
```

Let me know if those `span`s were there on purpose and we want them back.

Reply:
Check that there is no style being applied somewhere to `productName > span` or `h1 > span`, etc. If not, then this is probably OK to remove!